### ISSUE TITLE
FEAT Include GMX Spam Score

### ISSUE THREAD

**Issue by RalfRoemling** [friedPotat0/Spam-Scores#8]
Hi Overthere,
great plugin thanks alot :-)

One suggestion: Is it possible to implement other spamscores as well, e.g. the one from GMX?
```
X-GMX-Antispam: 6 (nemesis text pattern profiler); Detail=V3;
X-Spam-Flag: YES
```

**Comment by friedPotat0** [comment]
Thanks for opening this issue!

It seems that the scale is different for these headers. The values seem to consist only of positive values starting with 0 (no spam). Currently all mails with a score of 0 are marked yellow in the add-on. This would no longer apply to this scale.

First of all, the general score range customisability in the add-on options would have to be changed. I will keep it in mind for future versions, but I cannot promise that these headers will be supported very soon.

**Comment by RalfRoemling** [comment]
Yep, I know that the scales are different but because I changed my non-spam-rating to 0 to +2 I'm in the green :-)

But  I understand what you're pointing at. Maybe one could modularize antispam-tools and their individual ranges? Eventually by the users themselves or as a plugin-architecture? I'm no programmer so I can't guess what kind of trouble this would be...

**Comment by friedPotat0** [comment]
As soon as I have a little more spare time, I will consider a general solution for additional spam headers. There are more and more requests for alternative headers.

One could certainly define the spam header score ranges as some kind of preset, so that a standard range is given based on the respective header, but it can be adjusted at any time.

**Comment by friedPotat0** [comment]
By looking at the search results when searching for "X-GMX-Antispam" it seems that GMX does not have a spam scale at all. The number written to the header is just the reason it is recognised as spam. Zero means "mail was not recognized as spam" and all other numbers refer to a specific spam reason.

![image](https://user-images.githubusercontent.com/5374007/106836669-e3095080-6699-11eb-987c-54abc3ad0488.png)

All the add-on can do, is to display the red or green icon with a label "Spam" or "Ham". But no score can be displayed at all.

![image](https://user-images.githubusercontent.com/5374007/106842010-7dba5d00-66a3-11eb-9936-ab488fe0ee17.png)

Let me know if this is what you were looking for. Otherwise, I would be very happy to discuss alternative solutions.

**Comment by RalfRoemling** [comment]
yep, after some testing here seems you're right about not summing up. 

What you think of adding the number of the reason in the column nonetheless? It would make identifiying patterns in the overview easier. (But in case it's to much hassle it's good as it is i quess :-)

**Comment by friedPotat0** [comment]
Actually, it's easier to show the reason number in the column instead of making a distinction between GMX and all other score patterns. But I thought it could be very confusing for people who have several mail accounts from different services forwarding to the same inbox.

Maybe something like "Spam (1)" would be more appropriate to make clear that even lower numbers mean that the mail is classified as spam.

**Comment by RalfRoemling** [comment]
Hm, while rereading the thread I noticed that we missed sth.: 
Your idea would work fine for GMX alone but if the intention was to implement support for multiple antispamtools and so to broaden the plugins usebase, this would lead to more trouble with other providers. 

When thinking in this direction one would have to adapt the values in the column altogether, cause different tools would have different ranges for no-/spam indication. 

Eventually one would add the technology used? e.g. sth like this:
GMX 2 (red)
SA -24 (green)
SA 1.5 (yellow)
SA 5 (red)
rspamd 5 (red)
EOP SCL 5 (yellow)
EOP SCL 0 (green)
EOP BCL 5 (yellow)
EOP BCL ... and so forth

Don't know wether this isn't  a bit shootin' for the moon actually. It's your baby so it's your decision :-)

**Comment by friedPotat0** [comment]
Thanks for the feedback. I guess I won't implement GMX for the time being and instead work on a fine-tuned setting option in the add-on settings. I thought I could solve the GMX problem quickly, but there's actually more to it than I thought.

I'll get back as soon as I've managed to find a way of implementing it in a more general way.

**Comment by friedPotat0** [comment]
Revisited this as part of adding per-filter score handling. You're right that `X-GMX-Antispam` is a reason code, not a cumulative score, so it can't be thresholded like SpamAssassin. On master, GMX is now handled as a "flag" family: `0` shows as ham, any non-zero reason code shows as spam (matching `X-Spam-Flag`). The raw code is still shown as the value. Ships in the next release.

**Comment by RalfRoemling** [comment]
Thx a lot for the follow up and integration 👍
